### PR TITLE
define segment store repair method

### PR DESCRIPTION
**Pull request by cam-a**
This method will repair an injured segment passed in from the repair queue

### CODE REVIEW

File: pkg/storage/segments/store.go [hunk 1]
```diff
@@ -259,6 +260,12 @@ func (s *segmentStore) Delete(ctx context.Context, path paths.Path) (err error)
 	return s.pdb.Delete(ctx, path)
 }
 
+// Repair retrieves an at-risk segment and repairs and stores lost pieces on new nodes
+func (s *segmentStore) Repair(ctx context.Context, seg *pb.InjuredSegment) (err error) {
+	fmt.Println(seg)
+	return nil
```
Comment (egonelbre) [inline pkg/storage/segments/store.go]: `return errs.New("unimplemented")`

File: pkg/storage/segments/store.go [hunk 2]
```diff
@@ -259,6 +260,12 @@ func (s *segmentStore) Delete(ctx context.Context, path paths.Path) (err error)
 	return s.pdb.Delete(ctx, path)
 }
 
+// Repair retrieves an at-risk segment and repairs and stores lost pieces on new nodes
+func (s *segmentStore) Repair(ctx context.Context, seg *pb.InjuredSegment) (err error) {
```
Comment (egonelbre) [inline pkg/storage/segments/store.go]: The segment store shouldn't be responsible for repairing the segment. The repairer should do that an afterwards update the segment information.

Comment (aleitner) [reply pkg/storage/segments/store.go]: That contradicts this document https://docs.google.com/document/d/17LXgm8vMI0PyZmZ9PpJSl1qDyDxPosQf38A7qEFoYxw/edit#

**Comment by cam-a** [comment]
moved to https://github.com/storj/storj/pull/455

**Pull request closed without merging**